keras/src/backend/mlx/math.py

```diff
@@ -6,11 +6,37 @@
 
 
 def segment_sum(data, segment_ids, num_segments=None, sorted=False):
-    raise NotImplementedError("segment_sum is not implemented for mlx")
+    data = convert_to_tensor(data)
```

Comment:
mlx does not have unique and boolean indices  https://github.com/ml-explore/mlx/issues/246  and https://github.com/ml-explore/mlx/issues/377 . So the implementation is buggy with varying values of num_segments and negative segment_ids.

Reply:
One way to solve this is using numpy.unique on segment_ids to determine num_segments.

Reply:
@angeloskath could you please advise here?

Reply:
Hi @angeloskath Any update on above [comment](https://github.com/keras-team/keras/pull/19652#discussion_r1588632247) from @fchollet ? Please. Thank you!